src/main/java/org/curioswitch/common/protobuf/json/MessageMarshaller.java

```diff
@@ -383,22 +422,27 @@ public Builder sortingMapKeys(boolean sortingMapKeys) {
      * will not be usable with the returned {@link MessageMarshaller}.
      */
     public MessageMarshaller build() {
-      Map<Descriptor, TypeSpecificMarshaller<?>> builtParsers = new HashMap<>();
-      addStandardParser(BoolValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(Int32ValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(UInt32ValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(Int64ValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(UInt64ValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(StringValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(BytesValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(FloatValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(DoubleValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(TimestampMarshaller.INSTANCE, builtParsers);
-      addStandardParser(DurationMarshaller.INSTANCE, builtParsers);
-      addStandardParser(FieldMaskMarshaller.INSTANCE, builtParsers);
-      addStandardParser(StructMarshaller.INSTANCE, builtParsers);
-      addStandardParser(ValueMarshaller.INSTANCE, builtParsers);
-      addStandardParser(ListValueMarshaller.INSTANCE, builtParsers);
+      Map<Descriptor, TypeSpecificMarshaller<?>> builtParsers;
+      if (!preBuiltParsers.isEmpty()) {
+        builtParsers = new HashMap<>(preBuiltParsers);
+      } else {
+        builtParsers = new HashMap<>();
+        addStandardParser(BoolValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(Int32ValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(UInt32ValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(Int64ValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(UInt64ValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(StringValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(BytesValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(FloatValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(DoubleValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(TimestampMarshaller.INSTANCE, builtParsers);
+        addStandardParser(DurationMarshaller.INSTANCE, builtParsers);
+        addStandardParser(FieldMaskMarshaller.INSTANCE, builtParsers);
+        addStandardParser(StructMarshaller.INSTANCE, builtParsers);
+        addStandardParser(ValueMarshaller.INSTANCE, builtParsers);
+        addStandardParser(ListValueMarshaller.INSTANCE, builtParsers);
+      }
 
       AnyMarshaller anyParser = new AnyMarshaller();
```

Comment:
should `AnyMarshaller` also go under the else clause above if there aren't any preBuiltParsers ?

Reply:
Nah it needs to reference the newly built parsers so is re-registered. I added a comment to prevent confusion, thanks for pointing it out